sdk/core/azure-core/src/main/java/com/azure/core/util/UrlBuilder.java

```diff
@@ -159,13 +181,60 @@ public UrlBuilder setQuery(String query) {
         return this;
     }
 
+    /**
+     * Clear the query that will be used to build the final URL.
+     *
+     * @return This UrlBuilder so that multiple setters can be chained together.
+     */
+    public UrlBuilder clearQuery() {
+        if (query.isEmpty()) {
+            return this;
+        }
+
+        query.clear();
+        return this;
+    }
+
     /**
      * Get the query that has been assigned to this UrlBuilder.
      *
      * @return the query that has been assigned to this UrlBuilder.
      */
     public Map<String, String> getQuery() {
-        return query;
+        // This contains a map of key=value query parameters, replacing
+        // multiple values for a single key with a list of values under the same name,
+        // joined together with a comma. As discussed in https://github.com/Azure/azure-sdk-for-java/pull/21203.
+        final Map<String, String> singleKeyValueQuery =
+            this.query.entrySet()
+                      .stream()
+                      .collect(Collectors.toMap(
+                            e -> e.getKey(),
+                            e -> {
+                                QueryParameter parameter = e.getValue();
+                                String value = null;
+
+                                if (parameter != null) {
+                                    // join all parameters with a comma.
+                                    // name=a&name=b&name=c will become
+                                    // name=a,b,c
+                                    value = String.join(",", parameter.getValuesList());
+                                }
+
+                                return value;
+                            }
+                        ));
+
+        return singleKeyValueQuery;
+    }
+
+    /**
+     * Get the query that has been assigned to this UrlBuilder, with {@link QueryParameter} as the
+     * value type.
+     *
+     * @return the query that has been assigned to this UrlBuilder.
+     */
+    public Map<String, QueryParameter> getQueryParameters() {
+        return this.query;
```

Comment:
I'm not sure this is necessary at this stage, simply because no where else do users encounter `QueryParameter` in the API.

Reply:
Removed along with associated tests.